tests/web/test_server.py

```diff
@@ -299,3 +301,72 @@ def test_server_timing_header(
 
 def test_exact_showcase_output(showcase_output: str) -> None:
     assert SHOWCASE_RESUME == showcase_output
+
+
+@pytest.mark.filterwarnings("ignore:Request.message is deprecated")
+@pytest.mark.filterwarnings("ignore:Exception ignored in")
+class TestWebHandler:
+    async def test_web_handler_basic_functionality(
+        self,
+        aiohttp_client: Any,
+        aiohttp_server: Any,
+    ) -> None:
+        # Set up a mock resume server
+        async def mock_resume_handler(request):
+            return Response(
+                text=json.dumps({"basics": {"name": "Test User", "label": "Developer"}})
+            )
+
+        # Create and start mock server
+        mock_app = Application()
+        mock_app.router.add_get("/resume.json", mock_resume_handler)
+        mock_server = await aiohttp_server(mock_app)
+
+        # Create WebHandler pointing to our mock server
+        destination = f"http://localhost:{mock_server.port}/resume.json"
+        refresh_interval = timedelta(seconds=1)
+        handler = WebHandler(destination, refresh_interval=refresh_interval)
+
+        # Create test client
+        client = await aiohttp_client(handler.app)
+
+        # Test initial fetch
+        resp = await client.get("/")
+        assert resp.status == HTTPStatus.OK
+        content = await resp.text()
+        assert "Test User" in content
+        assert "Developer" in content
+
+        # Test caching
+        first_response = content
+        resp = await client.get("/")
+        assert await resp.text() == first_response
+
+        # Test refresh after interval
+        await asyncio.sleep(refresh_interval.total_seconds() + 0.1)
+        resp = await client.get("/")
+        assert resp.status == HTTPStatus.OK
+        assert await resp.text() == first_response  # Should be same content
+
```

Comment:
and then here you can add `assert hitcount == 2`

Reply:
Yep, following the answer above.